src/components/Header/Header.js

```diff
@@ -1,7 +1,45 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
 import './Header.scss';
 
-const Header = () => {
-  return <header className="header">header</header>;
-};
+function Header() {
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+
+  useEffect(() => {
+    const token = localStorage.getItem('token');
+    setIsLoggedIn(!!token); // 토큰이 있으면 true, 없으면 false
+  }, []);
+
+  const handleLogout = () => {
+    localStorage.removeItem('token');
+    setIsLoggedIn(false);
+  };
+
+  const handleLoginClick = () => {
+    if (isLoggedIn) {
+      handleLogout();
+    }
+  };
```

Comment:
```suggestion
    const token = localStorage.getItem('token');

  useEffect(() => {
    setIsLoggedIn(!!token); // 토큰이 있으면 true, 없으면 false
  }, []);

  const handleLoginClick = () => {
    isLoggedIn ? localStorage.removeItem("token") : 로그인 구현 함수 or 로그인 페이지 이동 함수
    setIsLoggedIn(prev => !prev)
  };
```

처음 header가 마운트 됐을 때, 토큰이 있는지 없는지 여부에 따라 로그인을 했는지 안했는지
판단하는 로직은 매우 좋습니다!

다만 UI를 보니 isLoggedIn에 따라서 로그인, 로그아웃 텍스트만 변경되고
handleLoginClick이라는 함수 하나로 관리하고 있기에
위와 같이 로그인 연관 state에 따라서 함수를 구현하는 것이 좋을 것 같네요!